src/docs/tiered-storage.dox

```diff
@@ -0,0 +1,82 @@
+/*! @page tiered_storage Tiered Storage
+
+\warning The WiredTiger Tiered Storage support has not been developed, tested or tuned in
+production scenarios, and should be viewed as experimental code.
+
+@section tiered_background Background
+
+A tiered table is a single btree (possibly with multiple checkpoints) made
+up of blocks that reside in one or more files on the local file system and
+zero or more cloud objects (e.g., stored in S3). The btree spans this set of
+files and objects by encoding a block's location in the pointer that links
+blocks in the btree. A tiered btree is contrasted with a regular (non-tiered)
+btree that is contained in a single file. It is also different from an LSM
+(see @ref lsm) tree, which WiredTiger implements as multiple regular btrees,
+each in its own file.
+
+When in use, a tiered btree, like a regular btree, may have some
+recently used or modified data that resides in memory pages.
+This in-memory representation is the same between a tiered btree
+and a regular btree, it is only how data is stored on disk and in
+cloud objects that makes these btrees different.
+
+All writes (i.e., pages that are reconciled and written to storage by eviction
+or checkpoint) are written to a designated file, called the active file.
+All other files and objects that are part of the tiered table are read-only.
+Each object or file is given an object number, which appears as part of its
+name on disk, cloud and in metadata.
```

Comment:
I stumbled on "cloud" on first reading. Maybe "in cloud storage"?

Reply:
Fixed.